tools/clean/atlas/network_peering.go

```diff
@@ -73,3 +100,27 @@ func (c *Cleaner) deleteNetworkPeering(ctx context.Context, projectID string, pe
 		fmt.Println(text.FgBlue.Sprintf("\t\t\tRequested deletion of network peering %s", peer.GetId()))
 	}
 }
+
+func (c *Cleaner) deleteNetworkPeeringContainers(ctx context.Context, projectID string, containers []admin.CloudProviderContainer) {
+	for _, container := range containers {
+		queryArgs := admin.ListPeeringConnectionsApiParams{GroupId: projectID, ProviderName: container.ProviderName}
+		peerReply, _, err := c.client.NetworkPeeringApi.ListPeeringConnectionsWithParams(ctx, &queryArgs).Execute()
+		if err != nil {
+			fmt.Println(text.FgRed.Sprintf("\t\t\tFailed to fetch peers on provider %q: %s", *container.ProviderName, err))
+		}
+		for _, peer := range peerReply.Results {
+			if peer.ContainerId == *container.Id {
+				_, _, err := c.client.NetworkPeeringApi.DeletePeeringConnection(ctx, projectID, *peer.Id).Execute()
+				if err != nil {
+					fmt.Println(text.FgRed.Sprintf("\t\t\tFailed to delete network peer %q with orphan container %q for project %q: %s",
+						*peer.Id, *container.Id, projectID, err))
+				}
+			}
+		}
```

Comment:
This did the trick to remove the GCP network peering I created manually for testing.

I now think the issue is that when we do `ListPeeringConnections` without parameters, it is NOT returning all the peerings, but only the AWS ones:
https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/#tag/Network-Peering/operation/updatePeeringContainer

providerName | stringDefault: "AWS"Enum: "AWS" "AZURE" "GCP"Cloud service provider to use for this VPC peering connection.

> QUERY PARAMETERS
> ...
> providerName	
> string
> Default: "AWS"
> Enum: "AWS" "AZURE" "GCP"
> Cloud service provider to use for this VPC peering connection.

So I think the proper fix is to query with params for all providers supported at `ListNetworkPeering`. I would expect that way we would not have to deal with orphan containers at all.

cc @helderjs